tutorials/compiler-explorer-deployment.md

```diff
@@ -0,0 +1,219 @@
+
+4. Open PRs:
+   4.1. Open issues in Compiler Explorer repo. Example: <https://github.com/compiler-explorer/compiler-explorer/issues/6651>.
+
+   4.2.: Open PR with change from `bin/yaml/libraries.yaml` in <https://github.com/compiler-explorer/infra>. Commit example: <https://github.com/compiler-explorer/infra/commit/96d504ac2d39212d22fc3dc5f8c7be3909457d29>.
+
+   4.3.: Open PR with change from `etc/config/c++.amazon.properties` in <https://github.com/compiler-explorer/compiler-explorer>. Commit example: <https://github.com/compiler-explorer/compiler-explorer/commit/738fd3b0c5979677e56d3550526efe2eaaa6f2fe>.
+
+   4.4. Link both PRs to your CE issue and to your Beman issue.
+
+   4.5. Ping CE owners for review (first landed PR should be done in the infra repo).
+
+### Header-only library, but a processing step is required - example: `beman.iterator_interface`
+
+Instructions are similar to previous section, but a build command neededs to be run to properly generate final deployable header.
+
+For example, to deploy / install `beman.iterator_interface` on Compiler Explorer, we follow previous steps:
+
+* Updates in `infra/bin/yaml/libraries.yaml`:
+
+```yaml
+...
+beman_iterator_interface:
+  after_stage_script:
+  - sed -i 's/add_subdirectory(examples)//g' CMakeLists.txt
+  build_type: cmake
+  check_file: README.md
+  extra_cmake_arg:
+  - -DBUILD_TESTING=OFF
+  lib_type: static
+  package_install: true
+  method: nightlyclone
+  repo: bemanproject/iterator_interface
+  staticliblink:
+  - beman.iterator_interface
+  targets:
+  - main
+  type: github
+...
+beman_optional:
+  build_type: none
+```
+
+Notes:
+
+* `build_type: cmake`: Run a cmake command.
+* `extra_cmake_arg:`: Pass arguments to cmake, e.g. disable building tests (`-DBUILD_TESTING=OFF`).
+* `lib_type: static`: Generate a static library.
+* `package_install: true`: Install build artifacts.
+
+* Updates in `compiler-explorer/etc/config/c++.amazon.properties`:
+
+```yaml
+...
+libs=...:beman_iterator_interface:beman_optional:...
+...
+libs.beman_iterator_interface.name=beman.iterator_interface
+libs.beman_iterator_interface.versions=trunk
+libs.beman_iterator_interface.url=https://github.com/bemanproject/iterator_interface
+libs.beman_iterator_interface.packagedheaders=true
+libs.beman_iterator_interface.staticliblink=beman.iterator_interface
+libs.beman_iterator_interface.versions.trunk.version=trunk
+libs.beman_iterator_interface.versions.trunk.lookupversion=main
+...
+```
+
+* Same testing instructions as per [Integrating a header only library](#integrating-a-header-only-library).
+
+## Non header-only library
+
+Currently, we don't have usages for such Beman libraries. We'll update the cheatsheet with instructions from [compiler-explorer: Adding a new library](https://github.com/compiler-explorer/compiler-explorer/blob/main/docs/AddingALibrary.md) when the first Beman example will be available.
```

Comment:
```suggestion
Currently, there are no such usages for such Beman libraries. 
```

I would keep it simple, I don't think it is necessarily to mention that the guide will be updated when we have more stuff.